File: src/main/java/com/m/blog/domain/menu/service/MenuService.java
package com.m.blog.domain.menu.service;

import com.m.blog.domain.boardCollection.application.domain.BoardCollection;
import com.m.blog.domain.boardCollection.infrastructure.repository.BoardCollectionDslRepository;
import com.m.blog.domain.boardCollection.infrastructure.repository.BoardCollectionJpaRepository;
import com.m.blog.domain.menu.application.query.MenuQuery;
import com.m.blog.domain.menu.infrastructure.web.dto.MenuResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@Transactional(readOnly = true)
@RequiredArgsConstructor
public class MenuService implements MenuQuery {

    private final BoardCollectionJpaRepository boardCollectionJpaRepository;
    private final BoardCollectionDslRepository boardCollectionDslRepository;

    @Override
    public MenuResponse get(){
        return MenuResponse.of(
                boardCollectionJpaRepository.findAll()
                , boardCollectionDslRepository.getAggregationDtos()
        );
    }
}

File: src/main/java/com/m/blog/domain/menu/service/MenuService.kt
package com.m.blog.domain.menu.service

import com.m.blog.domain.boardCollection.application.port.persistence.GetBoardCollectionPort
import com.m.blog.domain.menu.infrastructure.web.dto.MenuResponse
import com.m.blog.domain.menu.application.usecase.MenuUsecase
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
open class MenuService(private val getBoardCollectionQuery: GetBoardCollectionPort) : MenuUsecase {

    override fun get(): MenuResponse {
        return MenuResponse.of(
                getBoardCollectionQuery.getBoardCollectionEntities(),
                getBoardCollectionQuery.getBoardAggregationDtos()
        )
    }
}
